x/interchainqueries/keeper/keeper_test.go

```diff
@@ -126,6 +127,367 @@ func (suite *KeeperTestSuite) TestRegisterInterchainQuery() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestUpdateInterchainQuery() {
+	var msg iqtypes.MsgUpdateInterchainQueryRequest
+	originalQuery := iqtypes.MsgRegisterInterchainQuery{
+		QueryType: iqtypes.InterchainQueryTypeKV,
+		Keys: []*iqtypes.KVKey{
+			{
+				Path: "somepath",
+				Key:  []byte("somedata"),
+			},
+		},
+		TransactionsFilter: "",
+		ZoneId:             "osmosis",
+		ConnectionId:       suite.Path.EndpointA.ConnectionID,
+		UpdatePeriod:       1,
+		Sender:             "",
+	}
+
+	tests := []struct {
+		name              string
+		malleate          func(sender string)
+		expectedErr       error
+		expectedPeriod    uint64
+		expectedQueryKeys []*iqtypes.KVKey
+	}{
+		{
+			"valid update period",
+			func(sender string) {
+				msg = iqtypes.MsgUpdateInterchainQueryRequest{
+					QueryId:         1,
+					NewKeys:         nil,
+					NewUpdatePeriod: 2,
+					Sender:          sender,
+				}
+			},
+			nil,
+			2,
+			originalQuery.Keys,
+		},
+		{
+			"valid query data",
+			func(sender string) {
+				msg = iqtypes.MsgUpdateInterchainQueryRequest{
+					QueryId: 1,
+					NewKeys: []*iqtypes.KVKey{
+						{
+							Path: "newpath",
+							Key:  []byte("newdata"),
+						},
+					},
+					NewUpdatePeriod: 0,
+					Sender:          sender,
+				}
+			},
+			nil,
+			originalQuery.UpdatePeriod,
+			[]*iqtypes.KVKey{
+				{
+					Path: "newpath",
+					Key:  []byte("newdata"),
+				},
+			},
+		},
+		{
+			"valid query both query keys and update period",
+			func(sender string) {
+				msg = iqtypes.MsgUpdateInterchainQueryRequest{
+					QueryId: 1,
+					NewKeys: []*iqtypes.KVKey{
+						{
+							Path: "newpath",
+							Key:  []byte("newdata"),
+						},
+					},
+					NewUpdatePeriod: 2,
+					Sender:          sender,
+				}
+			},
+			nil,
+			2,
+			[]*iqtypes.KVKey{
+				{
+					Path: "newpath",
+					Key:  []byte("newdata"),
+				},
+			},
+		},
+		{
+			"invavid query id",
+			func(sender string) {
+				msg = iqtypes.MsgUpdateInterchainQueryRequest{
+					QueryId: 2,
+					NewKeys: []*iqtypes.KVKey{
+						{
+							Path: "newpath",
+							Key:  []byte("newdata"),
+						},
+					},
+					NewUpdatePeriod: 2,
+					Sender:          sender,
+				}
+			},
+			iqtypes.ErrInvalidQueryID,
+			originalQuery.UpdatePeriod,
+			originalQuery.Keys,
+		},
+		{
+			"failed due to empty request",
+			func(sender string) {
+				msg = iqtypes.MsgUpdateInterchainQueryRequest{
+					QueryId:         1,
+					NewKeys:         nil,
+					NewUpdatePeriod: 0,
+					Sender:          sender,
+				}
+			},
+			iqtypes.ErrInvalidSubmittedResult,
+			originalQuery.UpdatePeriod,
+			originalQuery.Keys,
+		},
+		{
+			"failed due auth error",
+			func(sender string) {
+				var (
+					ctx           = suite.ChainA.GetContext()
+					contractOwner = wasmKeeper.RandomAccountAddress(suite.T()) // We don't care what this address is
```

Comment:
I would prefer this comment to explain why we don't care. Currently, this comment only raises a big loud question.